File: Back/src/main/java/com/mutpio/pfe/Entities/RefBancaire.java
package com.mutpio.pfe.Entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;


@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class RefBancaire implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idRefBancaire;

    String frequence, iban, bic, titulaire;

    @Enumerated(EnumType.STRING)
    TypeRef typeRef;

    Integer jourPrelevement;

    @ManyToOne
    Devis devis;


}

File: Front/mutpio/src/app/Models/RefBancaire.ts
import {Devis} from "./Devis";
import {TypeRef} from "./TypeRef";

export class RefBancaire {
  idRefBancaire!: number;
  frequence!: string;
  iban!: string;
  bic!: string;
  titulaire!: string;
  typeRef!: TypeRef;
  jourPrelevement!: number;
  devis!: Devis;
}
